### PR TITLE
Update Rocstar test with OAK-blockchain v2.0.0

### PR DESCRIPTION
**Pull request by imstar15**
1. Reimplement `shibuyaHelper.getProxyAccount`.
2. Modify `WEIGHT_PROOF_SIZE` in `config/shibuya.js` and `config/rocstar.js`.
3. Use `shibuyaHelper.api.call.transactionPaymentApi.queryWeightToFee(weight)` to obtain the fee of a rocstar encoded call with native token(SBY, RSTR).

### CODE REVIEW

File: src/astar/rocstar.js [hunk 1]
```diff
@@ -142,8 +138,7 @@ const main = async () => {
 
     if (proxyBalance.free.lt(minBalance)) {
         console.log(`\nTopping up the proxy account on Shibuya with ${symbol} ...\n`);
-        const amount = new BN(100, 10);
-        const amountBN = amount.mul(decimalBN);
+        const amountBN = minBalance.muln(2);
```
Comment (chrisli30) [inline src/astar/rocstar.js:141]: What is this muln() function? I don’t see it from the bn.js library.

Comment (imstar15) [reply src/astar/rocstar.js:141]: https://github.com/indutny/bn.js/#postfixes

2 is a number.
`minBalance.muln(2)` is equal to `minBalance.mul(new BN(2))`

File: src/astar/rocstar.js [hunk 2]
```diff
@@ -126,7 +122,7 @@ const main = async () => {
     console.log(`\na) Add a proxy for ${account.name} If there is none setup on ${parachainName} (paraId:${turingHelper.config.paraId}) \n`);
 
     const proxyTypeParachain = 'Any'; // We cannotset proxyType to "DappsStaking" without the actual auto-restake call
-    const proxyOnParachain = shibuyaHelper.getProxyAccount(parachainAddress, turingHelper.config.paraId);
+    const proxyOnParachain = shibuyaHelper.getProxyAccount(turingAddress, turingHelper.config.paraId);
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/astar/rocstar.js:125]: Yes, It convert a location from account from Turing parachain to the Rocstar derived account.

[the remark this answers is not in the document]
Comment (imstar15) [reply src/astar/rocstar.js]: In fact, both of them will get the correct answer.

decodeAddress(address) will convert it to a accountId with ss58 format.

https://github.com/OAK-Foundation/xcm-demo/blob/886842d15f68086ec4bb25b6596395682a442cba/src/common/shibuyaHelper.js#L26

File: src/astar/rocstar.js [hunk 3]
```diff
@@ -6,68 +6,64 @@ import moment from 'moment';
...
-    const nextExecutionTime = (currentTimestamp - (currentTimestamp % millisecondsInHour)) / 1000 + secondsInHour;
+    const nextExecutionTime = getHourlyTimestamp(1) / 1000;
+    const timestampTwoHoursLater = getHourlyTimestamp(2) / 1000;
     const taskViaProxy = turingHelper.api.tx.automationTime.scheduleXcmpTaskThroughProxy(
-        { Recurring: { frequency: TASK_FREQUENCY, nextExecutionTime } },
-        // { Fixed: { executionTimes: [0] } },
-        shibuyaHelper.config.paraId,
-        paraTokenIdOnTuring,
+        { Fixed: { executionTimes: [nextExecutionTime, timestampTwoHoursLater] } },
+        { V3: { parents: 1, interior: { X1: { Parachain: shibuyaHelper.config.paraId } } } },
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/astar/rocstar.js:44]: I replaced multiple occurrences of assetLocation in the code with shibuyaHelper.getNativeAssetLocation().

File: src/common/shibuyaHelper.js [hunk 4]
```diff
@@ -166,6 +168,8 @@ class ShibuyaHelper {
         return location;
     };
 
+    getLocation = () => paraIdToLocation(this.config.paraId);
+
     getNativeAssetLocation = () => this.getAssetLocation(this.config.symbol);
```
Comment (chrisli30) [inline src/common/shibuyaHelper.js:173]: No, do not create a `symbol` for the chain config, which doesn’t really make sense. Let’s use the first element of the asset array as the native token.

so getNativeAssetLocation always returns the first symbol, and please remove the `symbol` in `this.config.symbol`

**Review by chrisli30 — CHANGES_REQUESTED** [review]

File: src/astar/rocstar.js (hunk 1 above)
Comment (chrisli30) [reply src/astar/rocstar.js:142]: Okay ... the thinking behind mulplying it by 2 is confusing and makes the code hard to read. If minBalance is the minimum requirement, why do we need to multiply it? Is `2` a special number? If you worry the balance is not enough, why not multiply by `10`?

**Review by chrisli30 — CHANGES_REQUESTED** [review]

File: src/common/shibuyaHelper.js (hunk 4 above)
Comment (chrisli30) [reply src/common/shibuyaHelper.js:173]: We decided not to update this part of the code since we plan to move xcm-config to oak.js

**Review by chrisli30 — APPROVED** [review]
Okay, looks good!

**Pull request merged**